### PR TITLE
refactor(rust/gui-client): begin isolating Tauri from our code

### PR DESCRIPTION
**Pull request by ReactorScram**
This moves about 2/3rds of the code from `firezone-gui-client` to `firezone-gui-client-common`.

I tested it in aarch64 Windows and cycled through sign-in and sign-out and closing and re-opening the GUI process while the IPC service stays running. IPC and updates each get their own MPSC channel in this, so I wanted to be sure it didn't break.

### CODE REVIEW

**Comment by ReactorScram** [comment]
@thomaseizinger I moved a few modules and it compiles fine, can we merge this or should I try to move all of them?

File: rust/gui-client/src-common/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+[package]
+name = "firezone-gui-client-common"
+# mark:next-gui-version
+version = "1.3.1"
```
Comment (thomaseizinger) [inline rust/gui-client/src-common/Cargo.toml:4]:
```suggestion
version = "0.1.0"
```

Does this need to track the GUI version?

File: rust/Cargo.toml [hunk 2]
```diff
@@ -7,7 +7,8 @@ members = [
   "connlib/shared",
   "connlib/snownet",
   "connlib/tunnel",
-  "gateway",
+  "gateway", "gui-client/src-common",
+  "gui-client/src-common",
```
Comment (thomaseizinger) [inline rust/Cargo.toml]: Can we have line-breaks here?

File: rust/gui-client/src-common/src/deep_link.rs [hunk 3]
```diff
@@ -133,7 +133,7 @@ mod tests {
         Ok(())
     }
 
-    fn parse_callback_wrapper(s: &str) -> Result<super::AuthResponse> {
+    fn parse_callback_wrapper(s: &str) -> Result<crate::auth::Response> {
```
Comment (thomaseizinger) [inline rust/gui-client/src-common/src/deep_link.rs]:
```suggestion
    fn parse_callback_wrapper(s: &str) -> Result<auth::Response> {
```

Seems to already be in scope.

**Review by thomaseizinger — APPROVED** [review]
LGTM

File: rust/gui-client/src-common/Cargo.toml (hunk 1 above)
Comment (ReactorScram) [reply rust/gui-client/src-common/Cargo.toml:4]: I thought it wouldn't hurt 🤷‍♀️

File: rust/Cargo.toml (hunk 2 above)
Comment (ReactorScram) [reply rust/Cargo.toml]: yeah idk how that happend

**Pull request merged**